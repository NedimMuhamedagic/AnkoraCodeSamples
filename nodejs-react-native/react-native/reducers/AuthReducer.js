import {
  LOGIN_USER_FAIL,
  LOGIN_USER_START,
  LOGIN_USER_SUCCESS,
} from '../actions/types'

export const INITIAL_STATE = {
  loading: false,
  error: '',
  user: null,
}

export default ( state = INITIAL_STATE, action ) => {
  switch ( action.type ) {

    case LOGIN_USER_START:
      return { ...state, loading: true, error: '' }

    case LOGIN_USER_SUCCESS:
      return {
        ...state,
        ...INITIAL_STATE,
        user: action.payload,
      }

    case LOGIN_USER_FAIL:
      return { ...state, error: 'Authentication failed', loading: false }

    default:
      return state
  }
}
