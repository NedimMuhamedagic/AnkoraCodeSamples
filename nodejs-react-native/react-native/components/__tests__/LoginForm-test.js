import React from 'react'

import LoginForm from '../LoginForm'
import AuthReducer, { INITIAL_STATE } from '../../reducers/AuthReducer'

import {
  LOGIN_USER_FAIL,
  LOGIN_USER_START,
  LOGIN_USER_SUCCESS,
} from '../../actions/types'

const shapeState = ( state ) => ( { auth: state } )

test( 'INITIAL_STATE', () => {
  const wrapper = shallowWithStore( <LoginForm />, shapeState( INITIAL_STATE ) )
  expect( wrapper.dive() ).toMatchSnapshot()
} )

test( 'LOGIN_USER_START', () => {
  const startState = AuthReducer( INITIAL_STATE, { type: LOGIN_USER_START } )
  const wrapper = shallowWithStore( <LoginForm />, shapeState( startState ) )
  expect( wrapper ).toMatchSnapshot()
} )

test( 'LOGIN_USER_FAIL', () => {
  const errorState = AuthReducer( INITIAL_STATE, { type: LOGIN_USER_FAIL } )
  const wrapper = shallowWithStore( <LoginForm />, shapeState( errorState ) )
  expect( wrapper ).toMatchSnapshot()
} )

test( 'LOGIN_USER_PASS', () => {
  const successState = AuthReducer( INITIAL_STATE, { type: LOGIN_USER_SUCCESS, payload: { foo: 'bar' } } )
  const wrapper = shallowWithStore( <LoginForm />, shapeState( successState ) )
  expect( wrapper ).toMatchSnapshot()
} )
