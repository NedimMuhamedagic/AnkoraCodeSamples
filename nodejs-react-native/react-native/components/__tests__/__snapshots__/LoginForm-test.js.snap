// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INITIAL_STATE 1`] = `
<Card>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Continue with Google+
  </Text>
  <branch(LoginButton)
    loading={false}
    onButtonPress={[Function]}
  />
  <FormValidationMessage />
</Card>
`;

exports[`LOGIN_USER_FAIL 1`] = `
<LoginForm
  error="Authentication failed"
  loading={false}
  loginWithGoogle={[Function]}
/>
`;

exports[`LOGIN_USER_PASS 1`] = `
<LoginForm
  error=""
  loading={false}
  loginWithGoogle={[Function]}
/>
`;

exports[`LOGIN_USER_START 1`] = `
<LoginForm
  error=""
  loading={true}
  loginWithGoogle={[Function]}
/>
`;
